### PR TITLE
IBX-2867: Fixed custom tags when defined without attributes

### PR DESCRIPTION
**Pull request by wizhippo**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [IBX-2867](https://issues.ibexa.co/browse/IBX-2867)
| **Bug**| yes
| **New feature**    | no
| **Target version** | 4,0
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

Check if customAttributes exist and are an array before trying to iterate.

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: src/bundle/Resources/public/js/CKEditor/custom-attributes/ui/custom-attributes-form-view.js [hunk 1]
```diff
@@ -146,20 +146,22 @@ class IbexaCustomAttributesFormView extends View {
             children.add(classesView);
         }
 
-        Object.entries(customAttributes).forEach(([name, config]) => {
-            const createAttributeMethod = this.attributeRenderMethods[config.type];
+        if (customAttributes && Object.keys(customAttributes).length !== 0) {
```
Comment (dew326) [inline src/bundle/Resources/public/js/CKEditor/custom-attributes/ui/custom-attributes-form-view.js:149]:
```suggestion
        if (customAttributes) {
```
I think this check is enough

**Review by dew326 — COMMENTED** [review]
Hi @wizhippo, thanks for the contribution.

This bug is in 4.1 as well, so the target branch should be 4.1. Please change it and rebase your branch.

**Review by dew326 — APPROVED** [review]

**Review by GrabowskiM — APPROVED** [review]

**Review by lucasOsti — APPROVED** [review]

**Review by tischsoic — APPROVED** [review]

**Review by katarzynazawada — CHANGES_REQUESTED** [review]
Unfortunately, I tested this fix and issue still occurs.
Zrzut ekranu 2022-05-12 o 10 51 49

I added following configuration 
```
ibexa:
    system:
        default:
            fieldtypes:
                ezrichtext:
                    custom_tags: [ezfactbox]
                    toolbar:
                        custom_tags_group:
                            buttons:
                                ezfactbox:
                                    priority: 5
ibexa_fieldtype_richtext:
     custom_tags:
        ezfactbox:
            template: field_type/ibexa_fieldtype_richtext/custom_tags/ezfactbox.html.twig
            icon: '/assets/field_type/ibexa_fieldtype_richtext/custom_tags/icon/factbox.svg#factbox'```

**Comment by wizhippo** [comment]
Found additional location where check is needed,

**Comment by wizhippo** [comment]
Found additional location where check is needed

**Review by dew326 — APPROVED** [review]

**Review by katarzynazawada — APPROVED** [review]

**Comment by davidsayre** [comment]
I'm using DXP 4.3 and this bug is appearing.
I have a simple 'separator' and this creates a richtext editor when inserted in

ibexa_fieldtype_richtext:
        separator:
            template: '@standard/ezrichtext/custom_tags/separator.html.twig'
            icon: '/bundles/ibexaicons/img/all-icons.svg#interface-block'
            # ezcontent field is implied / has no attributes

**Comment by fsoedjede** [comment]
Hello @wizhippo,

I'm using DXP 4.3 and I have a bug when I don't define attributes. Same case as @davidsayre .

After investigating, I found that the problem can be fixed at two places:

1. In PHP class

Add empty attributes item in the array
https://github.com/ibexa/fieldtype-richtext/blob/main/src/lib/Configuration/UI/Mapper/CustomTag.php#L95-L99
```diff
            $config[$tagName] = [
                'label' => "ezrichtext.custom_tags.{$tagName}.label",
                'description' => "ezrichtext.custom_tags.{$tagName}.description",
                'isInline' => $customTagConfiguration['is_inline'],
+               'attributes' => [],
            ];
```

*It's the solution I applied in my use case*

2. In JS files

- https://github.com/ibexa/fieldtype-richtext/blob/main/src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js#L173-L177 **AND** https://github.com/ibexa/fieldtype-richtext/blob/main/src/bundle/Resources/public/js/CKEditor/custom-tags/inline-custom-tag/inline-custom-tag-ui.js#L115-L119
```diff
-        const values = Object.entries(this.config.attributes).reduce((outputValues, [attributeName, config]) => {
-            outputValues[attributeName] = config.defaultValue;
-
-            return outputValues;
-        }, {});
+        let values = {};
+        if (this.config.attributes !== undefined) {
+            values = Object.entries(this.config.attributes).reduce((outputValues, [attributeName, config]) => {
+                outputValues[attributeName] = config.defaultValue;
+
+                return outputValues;
+            }, {});
+        }
```

Explanation:

As `attributes` is not provided in some case, the error occures because some parts of JS assume that there is always attributes.

The suggested changes are an attempt to fix the issues I saw.

Can I open a PR for that?

**Comment by wizhippo** [comment]
@fsoedjede Of course you can open PR for this.  Glad you found a solution.

**Pull request merged**